packages/python-google-compute-engine/google_compute_engine/instance_setup/instance_setup.py

```diff
@@ -140,6 +153,27 @@ def _GenerateSshKey(self, key_type, key_dest):
 
     file_utils.SetPermissions(key_dest, mode=0o600)
     file_utils.SetPermissions('%s.pub' % key_dest, mode=0o644)
+    with open('%s.pub' % key_dest, 'r') as pk:
+        key_data = pk.read()
+
+    key_values = key_data.split()
+    if len(key_values) < 2:
+        return
+    else:
+        return key_values[0], key_values[1]
+
+  def _WriteHostKeyToGuestAttributes(self, key_type, key_value):
```

Comment:
We can make a new module for writing metadata since this is generally usable. It's fine to leave it as-is until the feature is GA.

Reply:
Yes, as we discussed, it's a good idea to split this out, but probably better to wait until GA. Once guest attributes is GA we can create a module for writing those.